libvast/src/system/partition.cpp

```diff
@@ -387,6 +387,10 @@ caf::error unpack(const fbs::partition::v0& x, partition_synopsis& ps) {
     return caf::make_error(ec::format_error, "missing partition synopsis");
   if (!x.type_ids())
     return caf::make_error(ec::format_error, "missing type_ids");
+  // The id_range was only added in VAST 2021.08.26, so we fill it
+  // from the data in the partition if it does not exist.
+  if (!x.partition_synopsis()->id_range())
+    return unpack(*x.partition_synopsis(), ps, x.offset(), x.events());
```

Comment:
Wouldn't just assigning them suffice?
```suggestion
  if (!x.partition_synopsis()->id_range()) {
    ps.offset = x.offset();
    ps.events = x.events();
  }
```

Reply:
I had that version initially, but it also requires permanently removing the check for `id_range` in the regular `unpack()` function, which would need to allow "invalid" flatbuffers trusting that the caller fixes it up later. Which felt less clean than making a separate overload for this.